### ISSUE TITLE
Image previews and nginx/third party locations

### ISSUE THREAD

**Issue by tebba-von-mathenstein** [django-fiber/django-fiber#193]
I'm trying to use django-fiber for a project that serves static content from nginx, and occasionally from a 3rd party location. I've encountered an issue where image previews render as broken, but images on the front facing site work just fine.  

I believe isolated the code responsible, there are three places:

```
class AdminImageWidgetWithPreview(AdminFileWidget):
    def render(self, name, value, attrs=None):
        output = []
        if value and isinstance(value, ImageFieldFile):
            file_name = str(value)
            thumbnail = get_thumbnail(file_name, thumbnail_options=DETAIL_THUMBNAIL_OPTIONS)
            if thumbnail:
                try:
                    output.append('<img src="{0}" width="{1}" height="{2}" />'.format(thumbnail.url, thumbnail.width, thumbnail.height))
                except IOError:
                    # original image file is missing
                    output.append(_('<p>{0}</p>').format('Image file is missing'))
        output.append(super(AdminImageWidgetWithPreview, self).render(name, value, attrs))
        return mark_safe(u''.join(output))
```

```
def preview(self):
        thumbnail = get_thumbnail(self.image, thumbnail_options=LIST_THUMBNAIL_OPTIONS)
        if thumbnail:
            try:
                return u'<img src="{0}" width="{1}" height="{2}" />'.format(thumbnail.url, thumbnail.width, thumbnail.height)
            except IOError:
                # original image file is missing
                return _('Image file is missing')
```

and 

```

class ImageSerializer(serializers.HyperlinkedModelSerializer):
    image_url = serializers.Field(source='image.url')
    thumbnail_url = serializers.Field(source='thumbnail_url')
    filename = serializers.Field(source='get_filename')
    size = serializers.Field(source='get_size')
    can_edit = CanEditField()
    updated = UpdatedField()

    class Meta:
        model = Image
        read_only_fields = ('created', )
```

Each of these three places (the image serializer via source=thumbnail_url) attempt to create a thumbnail from the location the file was originally uploaded to. Since we never serve our assets directly from djagno (and indeed we cannot guarantee a file will exist there in perpetuity) that thumbnail does not exist - the image DOES exist however at Image.url via django-storages.

I've applied a monkey patch that doesn't feel quite right but allows image previews to show up as I expect them to by removing the assumption that a thumbnail can be created from the file at that location:

```
class AdminImageWidgetWithPreview(AdminFileWidget):
    def render(self, name, value, attrs=None):
        output = []
        if value and isinstance(value, ImageFieldFile):
            try:
                output.append('<img src="{0}" width="{1}" height="{2}" />'.format(value.url, value.width, value.height))
            except IOError:
                # original image file is missing
                output.append(_('<p>{0}</p>').format('Image file is missing'))
        output.append(super(AdminImageWidgetWithPreview, self).render(name, value, attrs))
        return mark_safe(u''.join(output))


def preview(self):
        try:
            return u'<img src="{0}" width="{1}" height="{2}" />'.format(self.image.url, self.width, self.height)
        except IOError:
            # original image file is missing
            return _('Image file is missing')


class ImageSerializer(serializers.HyperlinkedModelSerializer):
    image_url = serializers.Field(source='image.url')
    thumbnail_url = serializers.Field(source='image.url')
    filename = serializers.Field(source='image.url')
    size = serializers.Field(source='get_size')
    can_edit = CanEditField()
    updated = UpdatedField()

    class Meta:
        model = fiber_models.Image
        read_only_fields = ('created', )


fiber_widgets.AdminImageWidgetWithPreview = AdminImageWidgetWithPreview
fiber_models.Image.preview = preview
fiber_serializers.ImageSerializer = ImageSerializer
```

Perhaps I've missed something critical, is it possible for Fiber to support an architecture which always wants to serve static resources via a third party, or via nginx? If not, is it something you'll consider supporting? 

Basically I want the ability to tell Fiber at image upload time "this image won't exist here on the filesystem anymore, but you can assume it exists at URL." Because fiber.models.Image.url is already the correct location. 

Thanks for maintaining this project and thanks in advance for your time!

EDIT: I created a gist for the monkey patch, in case anyone would rather view it with syntax highlighting etc.
https://gist.github.com/teb311/9496500
